eddsa/eddsa.go

```diff
@@ -72,14 +72,20 @@ func (e *EdDSA) MarshalBinary() ([]byte, error) {
 	return eddsa, nil
 }
 
+//UnmarshalBinary transforms a slice of bytes into a EdDSA signature
 func (e *EdDSA) UnmarshalBinary(buff []byte) error {
 	if len(buff) != 64 {
 		return errors.New("wrong length for decoding EdDSA private")
 	}
 
 	e.seed = buff[:32]
-	e.prefix = buff[32:64]
-	e.Secret = suite.Scalar().SetBytes(e.seed)
+
+	scalar := sha512.Sum512(e.seed)
+	scalar[0] &= 0xf8
+	scalar[31] &= 0x3f
+	scalar[31] |= 0x40
```

Comment:
It would be best if you could factor that part out in a method and use it here and also in `NewEdDSA` so we dont' repeat code un-necessarily.